src/main/java/com/google/api/codegen/metacode/FieldStructureParser.java

```diff
@@ -78,52 +55,32 @@ public static InitCodeNode parse(String initFieldConfigString) {
 
   public static InitCodeNode parse(
       String initFieldConfigString, Map<String, InitValueConfig> initValueConfigMap) {
-    String fieldName = null;
-    String entityName = null;
-    String value = null;
-
-    String[] equalsParts = initFieldConfigString.split("[=]");
-    if (equalsParts.length > 2) {
-      throw new IllegalArgumentException("Inconsistent: found multiple '=' characters");
-    } else if (equalsParts.length == 2) {
-      value = equalsParts[1];
-    }
-
-    String[] fieldSpecs = equalsParts[0].split("[%]");
-    fieldName = fieldSpecs[0];
-    if (fieldSpecs.length == 2) {
-      entityName = fieldSpecs[1];
-    } else if (fieldSpecs.length > 2) {
-      throw new IllegalArgumentException("Inconsistent: found multiple '%' characters");
-    }
-    InitFieldConfig fieldConfig =
-        new AutoValue_FieldStructureParser_InitFieldConfig(
-            initFieldConfigString, fieldName, entityName, value);
-
-    InitValueConfig valueConfig = null;
-    if (fieldConfig.hasInitValue()) {
-      valueConfig = createInitValueConfig(fieldConfig, initValueConfigMap);
-    } else if (initValueConfigMap.containsKey(fieldConfig.fieldPath())) {
-      valueConfig = initValueConfigMap.get(fieldConfig.fieldPath());
-    }
-
+    InitFieldConfig fieldConfig = InitFieldConfig.of(initFieldConfigString);
+    InitValueConfig valueConfig =
+        createInitValueConfig(InitFieldConfig.of(initFieldConfigString), initValueConfigMap);
     return parsePartialDottedPathToInitCodeNode(
         fieldConfig.fieldPath(), InitCodeLineType.Unknown, valueConfig, null);
   }
 
   private static InitValueConfig createInitValueConfig(
       InitFieldConfig fieldConfig, Map<String, InitValueConfig> initValueConfigMap) {
-    if (fieldConfig.isFormattedField()) {
-      if (initValueConfigMap.containsKey(fieldConfig.fieldPath())) {
-        InitValueConfig valueConfig = initValueConfigMap.get(fieldConfig.fieldPath());
-        return valueConfig.withInitialCollectionValue(
-            fieldConfig.entityName(), fieldConfig.value());
+    InitValueConfig valueConfig = null;
+    if (fieldConfig.hasInitValue()) {
```

Comment:
Yes, I think that makes sense. Switched to use hasSimpleInitValue() and hasFormattedInitValue()